code/modules/mob/living/simple_animal/hostile/headcrab.dm

```diff
@@ -191,3 +198,363 @@
+			aggro_vision_range = gonarch_aggro_vision_range
+			vision_range = gonarch_standard_vision_range
+			speed = gonarch_standard_speed
+		if(homesick)
+			homesick = 0
+			step_towards(src, home_nest_turf, speed)
+		make_headcrabs() //Its its own proc, so they can be called by admins.
+	else
+		if(homesick >= 5)
+			mode_switch(GONARCH_MODE_RETURNING, gonarch_soundlist_return, change_icon = "gonarch_peaceful")
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_return()
+	speed = -1
+	var/turf/olddist = get_dist(src, home_nest_turf)
+	step_towards(src, home_nest_turf, speed)
+	target = home_nest_turf
+	if(get_dist(src, target) >= olddist)
+		frustration++
+		if(frustration > frustration_limit)
+			DestroyPathToTarget()
+			homesick++
+	else
+		frustration = 0
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_newhome()
+	var/list/turfs = list()
+	var/list/turfs_in_range = range(4, src)
+	for(var/turf/T in range(gonarch_nestfind_range, src))
+		if(T in turfs_in_range)
+			continue
+		if(isspaceturf(T))
+			continue
+		if(T.density)
+			continue
+		var/lightingcount = T.get_lumcount(0.5) * 10
+		if(lightingcount > 1 || gonarch_finding_home_attempts >= gonarch_finding_home_limit_attempts)
+			continue
+		turfs += T
+	if(!turfs.len)
+		gonarch_nestfind_range += 5
+		gonarch_finding_home_attempts++
+		return
+	home_nest_turf = pick(turfs)
+	home_nest_area = get_area(home_nest_turf)
+	mode_switch(GONARCH_MODE_RETURNING, gonarch_soundlist_return, change_icon = "gonarch_peaceful")
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_rampage()
+	//We want to lose all targets, and get the hell out of here. Vision/care is restored when back in base.
+	aggro_vision_range = 0
+	vision_range = 0
+	LoseTarget()
+	homesick = 0
+	for(var/mob/living/carbon/C in hearers(gonarch_screech_range, src))
+		to_chat(C, "<span class='warning'><font size='3'><b>You hear a ear piercing shriek and your senses dull!</font></b></span>")
+		C.Weaken(25)
+		C.MinimumDeafTicks(20)
+		C.Stuttering(20)
+		C.Stun(12)
+		C.Jitter(150)
+	var/i //Workaround to avoid compiler complaints and use faster for-loops
+	for(i in 1 to 4)
+		spawn_headcrabs(FALSE)
+	for(var/obj/structure/window/W in view(gonarch_screech_range))
+		W.deconstruct(FALSE)
+	mode_switch(GONARCH_MODE_NEWHOME, gonarch_soundlist_initialise, change_icon = "gonarch_peaceful")
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/spawn_headcrabs(spawn_as_children = TRUE)
+	var/type_to_spawn = pick(gonarch_spawn_list)
+	var/mob/living/simple_animal/hostile/headcrab/B = new type_to_spawn(get_turf(src))
+	var/chosen_sound = pick(gonarch_soundlist_birth)
+	playsound(src, chosen_sound, 200, TRUE, 8)
+	if(spawn_as_children)
+		LAZYADD(gonarch_children, B)
+		time_last_babies = world.time
+		visible_message("<span class='warning'>[B] crawls out of [src]!</span>")
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/make_headcrabs()
+	listclearnulls(gonarch_children)
```

Comment:
Make the headcrab remove itself from this list using signals.
`COMSIG_MOB_DEATH` is the signal you want to listen on the headcrab to see if it dies.